install/installer/traf_package_setup

```diff
@@ -0,0 +1,224 @@
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+# @@@ END COPYRIGHT @@@
+
+#  Setup Trafodion environment on a Cluster System
+
+#==============================================
+
+timestamp=$(date +%F-%H-%M-%S)
+source /etc/trafodion/trafodion_config
+
+cd $LOCAL_WORKDIR
+
+echo "***Log File: traf_package_setup script***" >> $INSTALL_LOG
+echo >> $INSTALL_LOG
+echo "Username: $TRAF_USER" >> $INSTALL_LOG
+echo "Nodes: $NODE_LIST" >> $INSTALL_LOG
+echo "Home directory: $HOME_DIR" >> $INSTALL_LOG
+echo >> $INSTALL_LOG
+
+echo "***INFO: Starting Trafodion Package Setup ($timestamp)"
+#========================================
+#Setting up pdsh variables
+if [ $node_count -eq 1 ]; then
+    TRAF_PDSH=""
+    TRAF_PDCP=""
+else
+    # use the -S option to cause pdsh to return largest of
+    # the remote command return values so we can tell if one
+    # or more of the remote commands failed
+    TRAF_PDSH="pdsh -R exec $MY_NODES ssh -q -n %h"
+    TRAF_PDCP="pdcp -R ssh $MY_NODES"
+fi
+#==========================================
+echo "***INFO: Installing required packages"
+echo "***INFO: Log file located in /var/log/trafodion"
+YUM_LOG=/var/log/trafodion/trafodion_yum_$timestamp.log
+ZYPP_LOG=/var/log/trafodion/trafodion_zypp_$timestamp.log
+
+#===============================
+#Set internet access
+internetAccess="true"
+
+#================================
+#Checking to see if epel package is installed. 
+if [[ $SUSE_LINUX == "false" ]]; then
+   for node in $NODE_LIST;
+   do
+     EPEL_INSTALLED=$(ssh -q -n $node rpm -qa | grep epel | wc -l)
+     if [[ $EPEL_INSTALLED == 0 ]]; then
+        break;
+     fi
+   done
+
+   if [[ $EPEL_INSTALLED == 0 ]]; then
+      echo "***INFO: ... EPEL rpm"
+      if [[ "$EPEL_RPM" != "" ]]; then
+         if [ $node_count -ne 1 ]; then
+            for node in $NODE_LIST
+            do
+               scp -q $EPEL_RPM $(whoami)@$node:$HOME
+            done
+         else
+            cp -rf $EPEL_RPM $HOME
+         fi
+      else
+         if [[ $internetAccess == "true" ]]; then
+            epel_rpm="epel-release-6-8.noarch.rpm"
+            wget http://download.fedoraproject.org/pub/epel/6/x86_64/$epel_rpm
+            if [ $node_count -ne 1 ]; then
+               for node in $NODE_LIST
+               do
+                  scp -q $LOCAL_WORKDIR/$epel_rpm $(whoami)@$node:$HOME
+               done
+            else
+               cp -rf $epel_rpm $HOME
+            fi
+
+            if [ $? != 0 ]; then
```

Comment:
What happens if scp fails on the first node but succeeds on the remaining ones? What would the value of $? be in that case?